src/main/java/com/microsoft/azure/relay/HybridConnectionListener.java

```diff
@@ -755,10 +780,10 @@ private void onOnline() {
 				this.lastError = null;
 				this.connectDelayIndex = -1;
 			}
-
 			RelayLogger.logEvent("connected", this.listener);
-			if (this.onlineHandler != null) {
-				this.onlineHandler.accept(this, null);
+			
+			if (this.listener.getOnlineHandler() != null) {
+				this.listener.getOnlineHandler().run();
```

Comment:
this pattern is not threadsafe. In cases like this the null check must check something that cannot become null between the check and the usage.

Reply:
Thanks, that's nice to know!

---
In reply to: [261311954](https://github.com/Azure/azure-relay-java/pull/26#discussion_r261311954) [](ancestors = 261311954)

Reply:
(The issue is someone could call setOnlineHandler(null) in-between the two getOnlineHandler calls in this snippet.)

---
In reply to: [261317780](https://github.com/Azure/azure-relay-java/pull/26#discussion_r261317780) [](ancestors = 261317780,261311954)